### PR TITLE
Fix interface name rule

### LINKED ISSUE #4624: interface IX123abc throw an error
### Bug Report

- TSLint V5.15.0:
- TypeScript V3.2.4:
- Running TSLint via: CLI, VSCode

#### TypeScript code being linted

```ts
interface II18nService {
}

class I18nService implements II18nService {
}
```

with `tslint.json` configuration:

```json
...
"interface-name": [
  true,
  "always-prefix"
],
...
```

#### Actual behavior
ERROR: .../i18n.service.ts: - interface name must start with a capitalized I

#### Expected behavior
No Error

**Examples that work:**
- IAnything
- IIanything
- INanything

Examples that don't work:
- II18anything
- IN18anything

It seems the numbers in the interface-name are a problem.

And by the way: A Class with the name I18nService works fine.

### PR DESCRIPTION
**Pull request by VincentLanglet**
#### PR checklist

- [x] Addresses an existing issue: fixes #4624
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests

#### Overview of change:
I stoped to use isLowerCase or isUpperCase to use a real regex. These functions was buggy with numbers.

#### Is there anything you'd like reviewers to focus on?

I add a better support for prettier too.

#### CHANGELOG.md entry:

 [bugfix] `interface-name` now correctly handle numeric characters

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -6,13 +6,7 @@
...
     "rules": {
         // Formatting rules
-        "prettier": {
-            "options": {
-                "printWidth": 100,
-                "tabWidth": 4,
-                "trailingComma": "all"
-            }
-        },
+        "prettier": true,
```
Comment (adidahiya) [inline tslint.json]: revert this part too

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (adidahiya) [inline .prettierrc]: please revert, don't make unrelated changes like this in PRs

Comment (VincentLanglet) [reply .prettierrc]: Done in another PR
